src/work/SimpleDAO.sol

```diff
@@ -111,75 +131,156 @@ contract SimpleDAO {
         require(block.timestamp <= proposal.expiryTime, "Proposal has expired");
         require(!proposal.voted[msg.sender], "Already voted");
 
         proposal.voted[msg.sender] = true;
 
+        uint votingPower = _getVotingPower(msg.sender);
         if (_vote) {
-            proposal.yesVotes += votingPower[msg.sender];
+            proposal.yesVotes += votingPower;
         } else {
-            proposal.noVotes += votingPower[msg.sender];
+            proposal.noVotes += votingPower;
         }
 
         emit Voted(_proposalId, msg.sender, _vote);
+
+        _updateProposalState(_proposalId);
     }
 
     /**
      * @dev Executes a proposal if it meets the quorum and voting threshold.
      * @param _proposalId The ID of the proposal to execute.
      */
-    function executeProposal(uint _proposalId) external onlyMember {
+    function executeProposal(uint _proposalId) external onlyMember notEmergencyStop {
         Proposal storage proposal = proposals[_proposalId];
+        require(proposal.proposalState == ProposalState.Succeeded, "Proposal has not succeeded");
+        require(block.timestamp > proposal.expiryTime + timelockDuration, "Timelock duration has not passed");
         require(!proposal.executed, "Proposal already executed");
-        require(block.timestamp <= proposal.expiryTime, "Proposal has expired");
-
-        uint participatingMembers = proposal.yesVotes + proposal.noVotes;
-        require(participatingMembers * 100 / totalMembers >= quorum, "Quorum not reached");
 
-        bool success = proposal.yesVotes * 100 / participatingMembers >= yesVoteThreshold;
-        if (success) {
-            proposal.executed = true;
-            (bool execSuccess,) = proposal.target.call(proposal.callData);
-            require(execSuccess, "Proposal execution failed");
-        }
+        proposal.executed = true;
+        (bool success,) = proposal.target.call{value: proposal.deposit}(proposal.callData);
+        require(success, "Proposal execution failed");
 
         emit ProposalExecuted(_proposalId, success);
     }
 
+    /**
+     * @dev Allows the contract owner to activate the emergency stop.
+     */
+    function activateEmergencyStop() external onlyOwner {
+        emergencyStop = true;
+        emit EmergencyStopActivated();
+    }
+
+    /**
+     * @dev Allows the contract owner to deactivate the emergency stop.
+     */
+    function deactivateEmergencyStop() external onlyOwner {
+        emergencyStop = false;
+        emit EmergencyStopDeactivated();
+    }
+
     /**
      * @dev Returns the total number of proposals.
      * @return The number of proposals.
      */
     function getProposalCount() external view returns (uint) {
         return proposals.length;
     }
-}
+
+    /**
+     * @dev Internal function to update the state of a proposal.
+     * @param _proposalId The ID of the proposal to update.
+     */
+    function _updateProposalState(uint _proposalId) internal {
```

Comment:
we need another method to manually defeat a proposal if no one votes on it before it expires.